### PR TITLE
Create 3.1 migration guide

### PR DESCRIPTION
**Pull request by jwm0**
This PR adds 3.1 migration guide with breaking changes related to Apps inside Dashboard, preview can be seen at: `/docs/3.0/developer/upgrade-guide/3-0-to-3-1`

### CODE REVIEW

File: docs/developer/upgrade-guide/3-0-to-3-1.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+title: Saleor 3.1
+---
+
+## Preparation
+
+The 3.1 version introduces some breaking changes that were necessary to ensure highest quality of our API and security. There were no changes in public APIs, so your storefronts do not need any changes to be compatible.
```
Comment (korycins) [inline docs/developer/upgrade-guide/3-0-to-3-1.mdx:7]: > There were no changes in public APIs, so your storefronts do not need any changes to be compatible.

Are you sure of this? I thought that we remove a lot of deprecated fields but also introduced a multichannel functionality.

Comment (jwm0) [reply docs/developer/upgrade-guide/3-0-to-3-1.mdx:7]: This is a migration guide for 3.1 (from 3.0), if there were any deprecated fields they would already be dropped in 3.0.

**Comment by krzysztofwolski** [comment]
@jwm0 Theres broken link preventing branch from building:
image

File: docs/developer/upgrade-guide/3-0-to-3-1.mdx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+### Manifest
+
+There are no breaking manifest changes, but Dashboard now uses `appUrl` for default App view. `configurationUrl` view can still be accessed from within Dashboard, but it's reserved for App's _configuration_ changes as the name suggests.
+
+### Extending Dashboard with Apps
+
+You can now use your Apps as extensions throughout the Dashboard. For more information and mounting points, see the the [documentation](https://docs.saleor.io/docs/3.0/developer/extending/apps/extending-dashboard-with-apps).
```
Comment (krzysztofwolski) [inline docs/developer/upgrade-guide/3-0-to-3-1.mdx:25]: @jwm0 please update the url: https://github.com/saleor/saleor-docs#lining-pages

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: docs/developer/upgrade-guide/3-0-to-3-1.mdx (hunk 2 above)
Comment (jwm0) [reply docs/developer/upgrade-guide/3-0-to-3-1.mdx]: Also fixed in 2.11 -> 3.0 migration guide

File: docs/developer/extending/apps/extending-dashboard-with-apps.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+## Key concepts
+
+App extensions allow extending different dashboard views by mounting the extension app inside a popup or redirecting users to the main app's page.
+The app is created as a secure iframe that protects against XSS attacks.
+It provides a way for customizing and extending the dashboard while keeping the code separate.
+
+## Manifest
+
+Apps can have multiple extensions. The extension is added by specifying it in the app's manifest.
+The below sample manifest defines two extensions that will be attached to the dashboard:
```
Comment (patrys) [inline docs/developer/extending/apps/extending-dashboard-with-apps.mdx:14]:
```suggestion
A single App can provide multiple extensions. Each extension is added by specifying it in the app's manifest.
The example manifest below defines two extensions, one providing a custom product action that opens a modal overlay, and a second one providing an alternative product creation page:
```

File: docs/developer/extending/apps/extending-dashboard-with-apps.mdx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      }
+    ],
+}
+```
+
+- `label` - Name which will be displayed in the dashboard
+- `mount` - Place, where extension will be mounted
+- `target` - Way of opening an extension's url. Default to `POPUP`.
+  `POPUP` will mount extension in an iframe view. `APP_PAGE` will redirect users to the dashboard's app page
+- `url` - Extension url. Url can start with `/` when `target` is set to `APP_PAGE` or manifest has defined `appUrl`.
```
Comment (patrys) [inline docs/developer/extending/apps/extending-dashboard-with-apps.mdx:46]:
```suggestion
- `label`: name which will be displayed in the dashboard
- `mount`: the place where the extension will be mounted
- `target`: method of presenting the interface (defaults to `POPUP`);
  `POPUP` will present the interface in a modal overlay, `APP_PAGE` will navigate to the application page
- `url`: URL of the view to display; you can skip the domain and protocol when `target` is set to `APP_PAGE`, or when your manifest defines an `appUrl`.
```

File: docs/developer/extending/apps/extending-dashboard-with-apps.mdx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  When `target` is set to `POPUP`, `url` will be used to render an iframe
```
Comment (patrys) [inline docs/developer/extending/apps/extending-dashboard-with-apps.mdx:47]:
```suggestion
  When `target` is set to `POPUP`, the `url` will be used to render an `<iframe>`.
```

File: docs/developer/extending/apps/extending-dashboard-with-apps.mdx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+## Possible mounting places
+
+Saleor requires that extension will define a place of mounting. The below table explains all mounting places available in Saleor.
```
Comment (patrys) [inline docs/developer/extending/apps/extending-dashboard-with-apps.mdx:51]:
```suggestion
Saleor requires extensions to define a mounting place. The table below explains all mounting locations currently supported by Saleor.
```

**Review by patrys — CHANGES_REQUESTED** [review]

File: docs/developer/upgrade-guide/3-0-to-3-1.mdx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+### Manifest
+
+There are no breaking manifest changes, but Dashboard now uses `appUrl` for default App view. `configurationUrl` view can still be accessed from within Dashboard, but it's reserved for App's _configuration_ changes as the name suggests.
```
Comment (karolkielecki) [inline docs/developer/upgrade-guide/3-0-to-3-1.mdx:21]:
```suggestion
There are no breaking manifest changes, but Dashboard now uses `appUrl` for the default app view. The `configurationUrl` view can still be accessed from within Dashboard, but it's reserved for the app's _configuration_ changes as the name suggests.
```

**Review by karolkielecki — CHANGES_REQUESTED** [review]

File: docs/developer/upgrade-guide/3-0-to-3-1.mdx (hunk 1 above)
Comment (korycins) [reply docs/developer/upgrade-guide/3-0-to-3-1.mdx:7]: Oh, sorry. My bad

**Review by korycins — APPROVED** [review]

**Review by karolkielecki — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Review by patrys — APPROVED** [review]

**Pull request merged**